src/effects/VST3/VST3Wrapper.cpp

```diff
@@ -222,6 +222,10 @@ class ComponentHandler : public Steinberg::Vst::IComponentHandler
    
    EffectSettings* mStateChangeSettings {nullptr};
    std::map<Steinberg::Vst::ParamID, Steinberg::Vst::ParamValue> mParametersCache;
+   std::map<Steinberg::Vst::ParamID, Steinberg::Vst::ParamValue> mCurrentParamValues;
+
+   bool mInStateChange { false };
```

Comment:
I think this variable isn't needed, and you can just use `mStateChangeSettings != nullptr`